controllers/application_controller.go

```diff
@@ -30,12 +31,32 @@ type ApplicationReconciler struct {
 // +kubebuilder:rbac:groups=orkestra.azure.microsoft.com,resources=applications/status,verbs=get;update;patch
 
 func (r *ApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("application", req.NamespacedName)
+	var requeue bool
+	var err error
+	var application orkestrav1alpha1.Application
 
-	// your logic here
+	ctx := context.Background()
+	logr := r.Log.WithValues("application", req.NamespacedName)
 
-	return ctrl.Result{}, nil
+	if err := r.Get(ctx, req.NamespacedName, &application); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		logr.Error(err, "unable to fetch Application")
+		return ctrl.Result{}, err
+	}
+
+	if application.Status.ChartStatus.Ready {
+		return ctrl.Result{}, nil
+	}
+
+	requeue, err = r.reconcile(logr, &application)
+	if err != nil {
+		logr.Error(err, "failed to reconcile application instance")
+		return ctrl.Result{}, err
```

Comment:
`ctrl.Result{Requeue: requeue}` here as well, as an error might be benign at times and we might still want to requeue